timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/LockAcquirer.java

```diff
@@ -83,6 +88,11 @@ private void acquireLocks() {
             try {
                 AsyncResult<Void> lockResult = AsyncResult.completedResult();
                 for (AsyncLock lock : locks.get()) {
+                    // This must happen BEFORE the lock is locked, it's safe to have a spurious lock.
+                    lockResult = lockResult.concatWith(() -> {
+                        lockEventProcessor.registerLock(lock.getDescriptor());
```

Comment:
for the caching perspective client side, is registering a lock despite not necessarily acquiring it, fine as long as the last unlock remains the same?

Reply:
That'll invalidate the old value. As discussed elsewhere it's probably worth actually putting it into `ExclusiveLock` though